registry/txt.go

```diff
@@ -157,11 +151,15 @@ func (im *TXTRegistry) Records(ctx context.Context) ([]*endpoint.Endpoint, error
 			dnsNameSplit[0] = im.wildcardReplacement
 		}
 		dnsName := strings.Join(dnsNameSplit, ".")
-		key := fmt.Sprintf("%s::%s", dnsName, ep.SetIdentifier)
-		if ep.RecordType == endpoint.RecordTypeAAAA {
-			key += keySuffixAAAA
+
+		// Handle both new and old registry format with the preference for the new one
+		key := createKey(dnsName, ep.SetIdentifier)
+		keyType := createKey(key, ep.RecordType)
+		labels, labelsExist := labelMap[keyType]
+		if !labelsExist {
+			labels, labelsExist = labelMap[key]
```

Comment:
It is incorrect for AAAA records to fall back to the old, untyped registry format. AAAA records never had ownership under the old format and they should not consume the data under the old ownership records for A records of the same domain.

Reply:
That is a good point. This should be good now.

Reply:
Issue is resolved.